pkg/system/phase4_configuring.go

```diff
@@ -33,6 +33,9 @@ func (r *Reconciler) ReconcilePhaseConfiguring() error {
 		"noobaa operator started phase 4/4 - \"Configuring\"",
 	)
 
+	if err := r.ReconcileNBSystem(); err != nil {
```

Comment:
I suggest to initiate these secrets with a bit different structure that uses ReconcileObject:

```golang
func (r *Reconciler) ReconcileSystemSecrets() error {

	if err := r.ReconcileObject(r.SecretAdmin, r.SetDesiredSecretAdmin); err != nil {
		return err
	}
	if err := r.ReconcileObject(r.SecretOp, r.SetDesiredSecretOp); err != nil {
		return err
	}

	r.NBClient.SetAuthToken(r.SecretOp.StringData["auth_token"])

	if err := r.ReconcileObject(r.SecretAdmin, r.SetDesiredSecretAdminAccountInfo); err != nil {
		return err
	}
	if err := r.ReconcileSecretEndpoints(); err != nil {
		return err
	}

	return nil
}


func (r *Reconciler) SetDesiredSecretAdmin() {
	r.SecretAdmin.StringData["system"] = r.NooBaa.Name
	r.SecretAdmin.StringData["email"] = options.AdminAccountEmail
	if r.SecretAdmin.StringData["password"] == "" {
		r.SecretAdmin.StringData["password"] = util.RandomBase64(16)
	}
}


func (r *Reconciler) SetDesiredSecretOp() {

	if r.SecretOp.UID != "" {
		// SecretOp exists means the system already created and nothing to do
		return
	}

	adminEmail := r.SecretAdmin.StringData["email"]
	adminPassword := r.SecretAdmin.StringData["password"]

	// If SecretOp is missing we need to create a system
	res, err := r.NBClient.CreateSystemAPI(nb.CreateSystemParams{
		Name:     r.Request.Name,
		Email:    adminEmail,
		Password: adminPassword,
	})
	if err == nil {
		r.SecretOp.StringData["auth_token"] = res.OperatorToken
		return
	}

	// If CreateSystem failed this is probably because the system already created before
	// but we haven't made it to write SecretOp
	res2, err := r.NBClient.CreateAuthAPI(nb.CreateAuthParams{
		System:   r.Request.Name,
		Role:     "admin",
		Email:    adminEmail,
		Password: adminPassword,
	})
	if err != nil {
		log.Errorf(...)
		return
	}
	r.NBClient.SetAuthToken(res2.Token)
	res3, err := r.NBClient.CreateAuthAPI(nb.CreateAuthParams{
		System: r.Request.Name,
		Role:   "operator",
		Email:  options.OperatorAccountEmail,
	})
	if err != nil {
		log.Errorf(...)
		return
	}
	r.SecretOp.StringData["auth_token"] = res3.Token
}


func (r *Reconciler) SetDesiredSecretAdminAccountInfo() {
	account, err := r.NBClient.ReadAccountAPI(nb.ReadAccountParams{
		Email: r.SecretAdmin.StringData["email"],
	})
	if err == nil && len(account.AccessKeys) > 0 {
		r.SecretAdmin.StringData["AWS_ACCESS_KEY_ID"] = account.AccessKeys[0].AccessKey
		r.SecretAdmin.StringData["AWS_SECRET_ACCESS_KEY"] = account.AccessKeys[0].SecretKey
	}
}


```